pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go

```diff
@@ -134,14 +140,28 @@ func addProxy(manifest *mf.Manifest) error {
 // transform mutates the passed manifest to one with common, component
 // and platform transformations applied
 func (r *Reconciler) transform(ctx context.Context, manifest *mf.Manifest, comp v1alpha1.TektonComponent) error {
-	//logger := logging.FromContext(ctx)
+	logger := logging.FromContext(ctx)
 	images := common.ToLowerCaseKeys(common.ImagesFromEnv(common.PipelinesImagePrefix))
 	instance := comp.(*v1alpha1.TektonPipeline)
+
 	extra := []mf.Transformer{
 		common.ApplyProxySettings,
 		common.DeploymentImages(images),
 		common.InjectLabelOnNamespace(),
 	}
+
+	// checks if configmaps already exists
+	cms, err := r.fetchConfigMaps(ctx, comp)
+	if err != nil {
+		logger.Errorf("failed to fetch configmaps from namespace %s", comp.GetSpec().GetTargetNamespace())
+		return err
+	}
+
+	// If configMaps exists, then preserve the configurations changes if any
+	if len(cms) != 0 {
+		extra = append(extra, common.PreserveExistingConfigMapsConfigurations(cms))
+	}
+
```

Comment:
could you move all of this logic into transformers, or to a separate file where transformers are located.
and just add `common.PreserveExistingConfigMapsConfigurations(cmNames)`